### DISCUSSION TITLE
Prioritize traces captured from LangGraph by relevance

### DISCUSSION THREAD

**Discussion by litagent** [langfuse/langfuse#1115] [Ideas]
### Describe the feature or potential improvement

Is there a way langfuse's CallbackHandler capture the necessary traces generated when using langgraph?
similar to langsmith screen shots attached. both langfuse & langsmith capture lot of unwanted information with null values which ideally needs to be omitted
![Screenshot_20240207_194239](https://github.com/langfuse/langfuse/assets/156183045/4cf80319-a594-48ea-9b7c-c137fe0e1177)
![Screenshot_20240207_194347](https://github.com/langfuse/langfuse/assets/156183045/7985b93a-6f27-4f09-aa85-f32c8840fa85)
![Screenshot_20240207_194322](https://github.com/langfuse/langfuse/assets/156183045/fc0b4c55-2bcd-4979-b253-3e11623a4fd4)

### Contribute

- [ ] Yes, I can implement this and raise a PR

**Comment by marcklingen** [comment]
Thanks for raising this idea of adding a `Most relevant` filter to get less verbose outputs in Langfuse. This seems to be a problem mostly related to the Langchain integration as other users of Langfuse define more deliberately what they'd like to add to a trace.

Is there a good heuristic of what's important/interesting to you in a Langgraph trace?

**Reply by phenomagentx** [reply]
LLM & tool related traces(start & end) are the most relavant for any agent implemented using Langchain or Langgraph @marcklingen

**Comment by hookenful** [comment]
Vouch for this

**Comment by marcklingen** [comment]
thanks again for adding this here, totally agree that some of the observations should rather be hidden by default.

in the meantime, @jannikmaierhoefer just contributed a notebook to the documentation with more LangGraph examples (just fyi): https://langfuse.com/docs/integrations/langchain/example-python-langgraph
